### ISSUE TITLE
Digitizing: rubberband is not deleted when editing mode is turned off

### ISSUE THREAD

**Issue by JuhoErvasti** [qgis/QGIS#58814]
### What is the bug or the crash?

When you're for example digitizing a new feature and you make the layer not editable the rubberband doesn't disappear:

![other_issue](https://github.com/user-attachments/assets/46bee3ec-7edf-40da-a00e-95adef7960c4)

### Steps to reproduce the issue

1. Create a polygon layer
2. Start adding a new layer
3. Turn layer editing off
4. Observe as the rubberband persists

### Versions

<table width='100%' align='center'><tr><td>QGIS version</td><td>3.39.0-Master</td></tr><tr><td>QGIS code revision</td><td><a href="https://github.com/qgis/QGIS/commit/d85f3c2a28">d85f3c2a28</a></td></tr><tr><td colspan="2">&nbsp;</td></tr><tr><td colspan="2"><b>Libraries</b></td></tr><tr><td>Qt version</td><td>5.15.3</td></tr><tr><td>Python version</td><td>3.10.12</td></tr><tr><td>GDAL/OGR version</td><td>3.6.4</td></tr><tr><td>PROJ version</td><td>9.1.1</td></tr><tr><td>EPSG Registry database version</td><td>v10.076 (2022-08-31)</td></tr><tr><td>GEOS version</td><td>3.11.1-CAPI-1.17.1</td></tr><tr><td>SQLite version</td><td>3.37.2</td></tr><tr><td>PDAL version</td><td>2.5.0</td></tr><tr><td>PostgreSQL client version</td><td>16.4 (Ubuntu 16.4-1.pgdg22.04+1)</td></tr><tr><td>SpatiaLite version</td><td>5.0.1</td></td></tr><tr><td>QWT version</td><td>6.1.4</td></tr><tr><td>QScintilla2 version</td><td>2.11.6</td></tr><tr><td>OS version</td><td>Ubuntu 22.04.4 LTS</td></tr><tr><td colspan="2"><i>This copy of QGIS writes debugging output.</i></td></tr><tr><td colspan="2">&nbsp;</td></tr><tr><td colspan="2"><b>Active Python plugins</b></td></tr><tr><td>processing</td><td>2.12.99</td></tr><tr><td>MetaSearch</td><td>0.3.6</td></tr><tr><td>db_manager</td><td>0.1.20</td></tr><tr><td>grassprovider</td><td>2.12.99</td></tr></table>

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

Doesn't occur in 3.38.2 Grenoble so looks like a regression. Other rubberband-related issue here: #58813

**Comment by JuhoErvasti** [comment]
Okay, well with more testing this is seems very inconsistent. I can't get it to reproduce now, and not entirely sure what if anything I did differently when this behavior happened...

**Comment by agiudiceandrea** [comment]
I also cannot replicate the issue.

**Comment by JuhoErvasti** [comment]
Yes, sorry I might've been too hasty in writing this up. I haven't really noticed a pattern here, it just randomly happened when I was investigating [this](https://github.com/qgis/QGIS/issues/58813).

Not sure, maybe this issue should be closed unless/until I figure out when it actually occurs?
